components/brave_wallet_ui/common/reducers/wallet_reducer.ts

```diff
@@ -387,4 +402,20 @@ reducer.on(WalletActions.setSitePermissions, (state: any, payload: SitePermissio
   }
 })
 
+reducer.on(WalletActions.queueNextTransaction, (state: any) => {
+  const pendingTransactions = sortTransactionByData(state.pendingTransactions)
+  const index = state.transactionQueueData.queueNumber
+  let newPendingTransaction = pendingTransactions[index]
+  let newQueueNumber = index + 1
+  if (pendingTransactions.length === index) {
+    newQueueNumber = 1
+    newPendingTransaction = pendingTransactions[0]
+  }
+  return {
+    ...state,
+    selectedPendingTransaction: newPendingTransaction,
+    transactionQueueData: { queueLength: pendingTransactions?.length ?? 0, queueNumber: newQueueNumber }
```

Comment:
Isn't `queueNumber` something that's unique per pending transaction? It's not clear which transaction `newQueueNumber` refers to. Also see my other remark related to `transactionQueueData`, because I'm not sure we need this field.